crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -1378,6 +1378,108 @@ unsafe impl<'__w, T: Component> QueryData for &'__w mut T {
     type ReadOnly = &'__w T;
 }
 
+/// Mirroring implementation for `&mut T` -> `&T` in a query, by providing `Mut<T>` -> `Ref<T>` for change detection in a query's read-only type.
```

Comment:
```
/// When `Mut<T`>` is used in a query, it will be converted to `Ref<T>` when transformed into its read-only form, providing access to change detection methods.
/// 
/// By contrast `&mut T` will result in a `Mut<T>` item in mutable form (to record mutations), but result in a bare `&T` in read-only form.
```

Reply:
Replace all docs above the safety comment with this.